src/commands/register.ts

```diff
@@ -147,7 +154,15 @@ export default class ComponentRegister extends Command {
       }
       let build_args: string[] = [];
       if (service_config.build?.args) {
-        build_args = Object.entries(service_config.build?.args).map(([key, value]) => `${key}=${value}`);
+        const build_args_map: Dictionary<string> = service_config.build?.args || {};
+        for (const arg of flags.arg || []) {
+          const [key, value] = arg.split('=');
+          if (!value) {
+            throw new Error('--arg must be in the format key=value');
+          }
+          build_args_map[key] = value;
+        }
+        build_args = Object.entries(build_args_map).map(([key, value]) => `${key}=${value}`);
```

Comment:
Can't we just set `build_args` straight to `service_config.build?.args || []` after passing validation? `build_args_map` doesn't seem necessary, though the validation does of course